emscripten.py

```diff
@@ -628,10 +628,16 @@ def optimize_syscalls(declares, DEBUG):
     # without filesystem support, it doesn't matter what syscalls need
     shared.Settings.SYSCALLS_REQUIRE_FILESYSTEM = 0
   else:
-    syscall_prefix = '__syscall'
-    syscall_numbers = [d[len(syscall_prefix):] for d in declares if d.startswith(syscall_prefix)]
-    syscalls = [int(s) for s in syscall_numbers if is_int(s)]
-    if set(syscalls).issubset(set([6, 54, 140])): # close, ioctl, llseek
+    syscall_prefixes = ('__syscall', 'fd_', '__wasi_fd_')
```

Comment:
why do we need `fd_read` in addition to `__wasi_fd_read`? Relying on/using non-underscored names seems error-prone.

Reply:
Also if we eventually support preopened FDs the way WASI imagines, then this will break things since these calls can work in the absence of `open`.